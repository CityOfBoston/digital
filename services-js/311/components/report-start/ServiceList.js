// @flow
import React from 'react';
import { css } from 'glamor';

import type { ServiceSummary } from '../../data/types';

const STYLES = {
  list: css({
    margin: '0.75em 0',
    padding: '0 1em',
    listStyle: 'none',
  }),

  item: css({
    ':before': {
      content: '""',
      borderColor: 'transparent #111',
      borderStyle: 'solid',
      borderWidth: '0.35em 0 0.35em 0.45em',
      display: 'block',
      height: 0,
      width: 0,
      left: '-1em',
      top: '0.9em',
      position: 'relative',
    },
  }),

  link: css({
    textDecoration: 'none',
    textTransform: 'uppercase',
    color: '#37a0e7',
    fontWeight: 500,
    fontSize: 15,
  }),
};

export default class ServiceList extends React.Component {
  props: {
    serviceSummaries: ServiceSummary[],
    onCodeChosen: (string) => void,
  }

  handleServiceClick = (ev: SyntheticInputEvent) => {
    ev.preventDefault();

    const code = ev.target.getAttribute('data-code');
    if (code) {
      this.props.onCodeChosen(code);
    }
  }

  render() {
    const { serviceSummaries } = this.props;
    return (
      <ul className={STYLES.list}>{ serviceSummaries.map(this.renderServiceButton) }</ul>
    );
  }

  renderServiceButton = ({ name, code }: ServiceSummary) => (
    <li className={STYLES.item} key={code}>
      <a className={STYLES.link} href={`/report/${code}/location`} data-code={code} onClick={this.handleServiceClick}>{name}</a>
    </li>
  );
}
