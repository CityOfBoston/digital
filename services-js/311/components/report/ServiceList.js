// @flow
import React from 'react';

import type { Service } from '../../store/modules/services';

export default class ServiceList extends React.Component {
  props: {
    services: ?Service[],
    servicesError: ?Object,
    onCodeChosen: (string) => void,
  }

  handleServiceClick = (ev: SyntheticInputEvent) => {
    ev.preventDefault();
    this.props.onCodeChosen(ev.target.value);
  }

  render() {
    const { services, servicesError } = this.props;

    if (servicesError) {
      return <h2>{servicesError.message}</h2>;
    }

    return (
      <ul>
        { services && services.map(this.renderServiceButton) }
      </ul>
    );
  }

  renderServiceButton = ({ name, code }: Service) => (
    <li key={code}>
      <button type="submit" name="code" value={code} onClick={this.handleServiceClick}>{name}</button>
    </li>
  );
}
