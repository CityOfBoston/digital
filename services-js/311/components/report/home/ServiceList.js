// @flow
import React from 'react';

import type { ServiceSummary } from '../../../data/types';

class ServiceButton extends React.Component {
  props: {
    serviceSummary: ServiceSummary,
    onServiceChosen: (code: string) => void,
  }

  whenClicked = (ev: SyntheticInputEvent) => {
    const { onServiceChosen, serviceSummary } = this.props;

    ev.preventDefault();
    onServiceChosen(serviceSummary.code);
  }

  render() {
    const { serviceSummary: { code, name } } = this.props;
    return (
      <li key={code}>
        <a className="t--sans tt-u" href={`/report/${code}`} onClick={this.whenClicked}>{name}</a>
      </li>
    );
  }
}

export type Props = {
  serviceSummaries: ServiceSummary[],
  onServiceChosen: (code: string) => void,
}

export default function ServiceList({ serviceSummaries, onServiceChosen }: Props) {
  return (
    <ul className="ul">{ serviceSummaries.map((s) => <ServiceButton key={s.code} serviceSummary={s} onServiceChosen={onServiceChosen} />) }</ul>
  );
}
