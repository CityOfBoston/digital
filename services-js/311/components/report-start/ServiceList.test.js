// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import ServiceList from './ServiceList';

const SERVICE_SUMMARIES = [{
  name: 'Needle Pickup',
  code: 'needles',
  hasMetadata: true,
}];

test('services', () => {
  const component = renderer.create(
    <ServiceList serviceSummaries={SERVICE_SUMMARIES} onCodeChosen={jest.fn()} />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
