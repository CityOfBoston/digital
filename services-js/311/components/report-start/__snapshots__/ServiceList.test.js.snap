exports[`test services 1`] = `
<ul
  className={
    Object {
      "data-css-117fchh": "",
    }
  }>
  <li
    className={
      Object {
        "data-css-14nloni": "",
      }
    }>
    <a
      className={
        Object {
          "data-css-3br7c8": "",
        }
      }
      data-code="needles"
      href="/report/needles/location"
      onClick={[Function]}>
      Needle Pickup
    </a>
  </li>
</ul>
`;
